cpp/simulations/munich_graph_sim.cpp

```diff
@@ -0,0 +1,297 @@
+    BOOST_OUTCOME_TRY(mio::set_population_data_age_group_names(population_data_names));
+
+    return mio::success();
+}
+
+mio::IOResult<void> set_nodes(mio::Graph<mio::osecir::Model, mio::MigrationParameters>& params_graph,
+                              const mio::osecir::Parameters& params, const fs::path& data_dir, mio::Date start_date)
+{
+    auto scaling_factor_infected = std::vector<double>(size_t(params.get_num_groups()), 1.0);
+    auto scaling_factor_icu      = 1.0;
+
+    BOOST_OUTCOME_TRY(set_age_group_names(
+        {"All"}, {"0-4 years", "5-9 years", "10-14 years", "15-19 years", "20-24 years", "25-29 years", "30-34 years",
+                  "35-39 years", "40-44 years", "45-49 years", "50-54 years", "55-59 years", "60-64 years",
+                  "65-69 years", "70-74 years", "75-79 years", "80-84 years", "85-89 years", "90+ years"}));
+    //read node ids
+    BOOST_OUTCOME_TRY(
+        node_ids,
+        mio::get_node_ids(
+            mio::path_join((data_dir).string(), "population_data.json"), false,
+            false)); //mio::path_join((data_dir).string(), "population_data.json") //"C:/Users/bick_ju/Documents/epidata/20230818/Germany/county_current_population.json"
+    std::vector<mio::osecir::Model> nodes(node_ids.size(), mio::osecir::Model(int(size_t(params.get_num_groups()))));
+    //set parameters for every node
+    for (auto& node : nodes) {
+        node.parameters = params;
+    }
+    auto read = mio::osecir::read_input_data(nodes, start_date, node_ids, scaling_factor_infected,
+                                                           scaling_factor_icu, data_dir.string(), 90, true);
+    for (size_t node_idx = 0; node_idx < nodes.size(); ++node_idx) {
+        params_graph.add_node(node_ids[node_idx], nodes[node_idx]);
+    }
+    return mio::success();
+}
+
+mio::IOResult<void> set_edges(mio::Graph<mio::osecir::Model, mio::MigrationParameters>& params_graph,
+                              const fs::path& data_dir)
+{
+    auto migrating_compartments = {mio::osecir::InfectionState::Susceptible, mio::osecir::InfectionState::Exposed,
+                                   mio::osecir::InfectionState::InfectedNoSymptoms,
+                                   mio::osecir::InfectionState::InfectedSymptoms,
+                                   mio::osecir::InfectionState::Recovered};
+    BOOST_OUTCOME_TRY(mobility_data,
+                      mio::read_mobility_plain(mio::path_join((data_dir).string(), "mobility_matrix.txt")));
+    if (mobility_data.rows() != Eigen::Index(params_graph.nodes().size()) ||
+        mobility_data.cols() != Eigen::Index(params_graph.nodes().size())) {
+        return mio::failure(mio::StatusCode::InvalidValue,
+                            "Mobility matrices do not have the correct size. You may need to run "
+                            "transformMobilitydata.py from pycode memilio epidata package.");
+    }
```

Comment:
I think this msg is useless in our usage, right ? :D

Reply:
The second part "You may need to run transformMobilitydata.py from pycode memilio epidata package." is useless, but the rest is useful, because we are checking whether the mobility matrices have as many rows/cols as we have nodes.